caf-audit-binding-elasticsearch/src/main/java/com/hpe/caf/auditing/elastic/ElasticAuditIndexManager.java

```diff
@@ -41,55 +39,46 @@ public final class ElasticAuditIndexManager {
 
     private ElasticAuditIndexManager(){}
 
-    public static void createIndexTemplate(final int numberOfShards, final int numberOfReplicas,
-                                           final RestHighLevelClient restHighLevelClient, final boolean isForceIndexTemplateUpdate)
+    public static void createTemplate(final int numberOfShards, final int numberOfReplicas,
+                                           final OpenSearchClient openSearchClient, final boolean isForceIndexTemplateUpdate)
         throws IOException
     {
-        if (isForceIndexTemplateUpdate || !isIndexTemplatePresent(restHighLevelClient)) {
-            final PutIndexTemplateRequest request = new PutIndexTemplateRequest(INDEX_TEMPLATE_NAME);
-            //  Configure the number of shards and replicas the new index should have.
-            final Settings indexSettings = Settings.builder()
-                .put("number_of_shards", numberOfShards)
-                .put("number_of_replicas", numberOfReplicas)
+        if (isForceIndexTemplateUpdate || !isTemplatePresent(openSearchClient)) {
+      
+            final Map<String, JsonData> indexSettings  = Map.of("number_of_shards", JsonData.of(numberOfShards),
+                                                          "number_of_replicas", JsonData.of(numberOfReplicas));
+            final PutTemplateRequest request = new PutTemplateRequest.Builder()
+                .name(INDEX_TEMPLATE_NAME)
```

Comment:
There has been name changes between the `RestHighLevelClient` and the `OpenSearchClient`

`PutIndexTemplateRequest` in old client = `PutTemplateRequest` in the new client

`PutComposableIndexTemplateRequest` in old client = `PutIndexTemplateRequest` in the new client

https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-templates-v1.html